src/config.rs

```diff
@@ -52,6 +52,18 @@ pub struct Config {
     extensions: Vec<ExtensionType>,
 }
 
+impl Config {
+    pub fn get_protocol_version_ref(&self) -> &[ProtocolVersion] {
```

Comment:
Do we need these? The global config can be accessed already with the functions below.
Also, if we want these functions we should put them in the same `impl` block.